Devscord.DiscordFramework/Framework/Commands/Services/BotCommandsPropertyConversionService.cs

```diff
@@ -25,17 +29,16 @@ public object ConvertType(string value, BotCommandPropertyType type)
 
         private TimeSpan ToTimeSpan(string value)
         {
-            var match = this._exTime.Match(value);
-            var unit = match.Groups["Unit"].Value.ToLowerInvariant();
-            var timeValue = uint.Parse(match.Groups["Value"].Value);
+            var matchGroups = this._exTime.Match(value).Groups;
+            var unit = matchGroups["Unit"].Value.ToLowerInvariant();
+            var timeValue = uint.Parse(matchGroups["Value"].Value);
             return unit switch
             {
                 "d" => TimeSpan.FromDays(timeValue),
                 "h" => TimeSpan.FromHours(timeValue),
                 "m" => TimeSpan.FromMinutes(timeValue),
                 "s" => TimeSpan.FromSeconds(timeValue),
-                "ms" => TimeSpan.FromMilliseconds(timeValue),
-                _ => throw new ArgumentNullException()
+                _ => TimeSpan.FromMilliseconds(timeValue),
```

Comment:
to zła praktyka -> ms powinny być tylko wtedy kiedy ktoś wybrał ms
myśle że domyślną wartością jeśli ktoś nie podał, powinny być jak już to minuty

ale najlepiej w ogóle nie łapać takich przypadków

Reply:
nie o to chodzi. Jeśli chodzi o ms jako default to dałem to przypadkowo ponieważ jak byś zobaczył dokładniej całą walidacje to można dość do wniosku, że jeśli dotarliśmy do tego momentu w parsowaniu to oznacza, że użytkownik podał czas z jednostką według regexu _exTime w BotCommandsMatchingService (czyli podał ms, s, m, h lub d bo tylko takie są możliwe), jeśli użytkownik nie podał jednostki lub dał coś innego to podczas walidacji zostanie to wyłapane i zostanie wyświetlony komunikat o niepoprawnym argumencie. Zamieniłem "ms" na _ tylko dlatego aby VS nie pokazywałby ostrzeżenia, że nie ma defaulta. Rozumiesz już?

Reply:
moim zdaniem i tak lepiej zostawić jak było wcześniej, czyli rzucanie wyjątku, ale może typu InvalidArgumentException, aby podwójnie się zabezpieczyć przed ewentualnym błędem w wcześniejszym kodzie

Reply:
Tylko, że do takiej sytuacji nigdy nie dojdzie bo user może tylko wybrać jedną z określonych jednostek a gdyby podał co innego to zostanie to wyłapane w walidacji. Poza tym parser nie powinien raczej robić takich rzeczy jak wywalanie wyjątków.

Reply:
wydaje mi się, że i tak lepiej pisać "ms" i jako default dać wyjątek
nawet z samego punktu przejrzystości - lepiej widać obsługiwane opcje

Reply:
to nadal nie jest resolved

Reply:
parser nie powinien raczej robić takich rzeczy jak wywalanie wyjątków

no jak jest wyjątek związany z parsowaniem to chyba jednak powinien go wyrzucić

Reply:
no dobra zrobię ten wyjątek tam jako default, ale kiedyś mówiłeś, że w parsowaniu nie powinno być wyjątków...

Reply:
wszystko zależy od tego czego one dotyczą